tests/integration/cmd_dev_test.go

```diff
@@ -303,6 +304,12 @@ var _ = Describe("odo dev command tests", func() {
 						ContainSubstring(fmt.Sprintf("%s-app-", cmpName)),
 					))
 				})
+
+				// Returned pvc yaml contains ownerreference
+				By("creating a pvc with ownerreference", func() {
+					output := commonVar.CliRunner.Run("get", "pvc", "--namespace", commonVar.Project, "-o", "yaml").Out.Contents()
+					Expect(string(output)).To(ContainSubstring("ownerReferences"))
```

Comment:
Can you be more specific here, by checking that this list of owner references does contain the Deployment? Here we are sure that there is an owner reference, but we can't tell for sure that it contains the one we want.
I think that was the actual issue: the PVC was not owned by the Deployment. 

Maybe using a JSONPath template expression like this can help:
```sh
kubectl get pvc --namespace <ns> \
  -o jsonpath='{range .items[*].metadata.ownerReferences[*]}{@..kind}{"/"}{@..name}{end}'
```
It will return both the kind and name of each owner reference separated by a `/`, e.g.: `Deployment/${componentName}-app`